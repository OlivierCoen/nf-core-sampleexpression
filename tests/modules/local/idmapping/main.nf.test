nextflow_process {

    name "Test Process IDMAPPING"
    script "modules/local/idmapping/main.nf"
    process "IDMAPPING"
    tag "idmapping"

    test("Map Ensembl IDs to themselves") {

        when {
            process {
                """
                count_file = file("$baseDir/tests/input/counts.ensembl_ids.csv")
                input[0] = [count_file, "Solanum tuberosum"]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.csv).match()
        }

    }

    test("Map NCBI IDs") {

        when {
            process {
                """
                count_file = file("$baseDir/tests/input/counts.ncbi_ids.csv")
                input[0] = [count_file, "Arabidopsis thaliana"]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.csv).match()
        }

    }


    test("Map Uniprot IDs") {

        when {
            process {
                """
                count_file = file("$baseDir/tests/input/counts.uniprot_ids.csv")
                input[0] = [count_file, "Arabidopsis thaliana"]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out.csv).match()
        }

    }

}
