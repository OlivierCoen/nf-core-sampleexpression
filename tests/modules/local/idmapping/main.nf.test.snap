{
    "Map Ensembl IDs to themselves": {
        "content": [
            [
                "counts.ensembl_ids_renamed.csv:md5,e4216c237b47e47162926e3d4a102728"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-15T12:20:49.941494532"
    },
    "Should run without failures": {
        "content": [
            [
                "counts.raw_renamed.csv:md5,a3259fd5524222138600a571dcacdf17"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-15T12:09:11.687854667"
    },
    "Map Uniprot IDs": {
        "content": [
            [
                "counts.uniprot_ids_renamed.csv:md5,cd88d623ff58193df380e88f772378a4"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-15T18:39:28.325984625"
    },
    "Map NCBI IDs": {
        "content": [
            [
                "counts.ncbi_ids_renamed.csv:md5,67a0af621122757dc46ceea88e57a922"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-15T12:23:28.179560555"
    }
}